test/functional/rpc_getblocklocations.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the getblocklocations rpc call."""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+class GetblocklocationsTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def run_test(self):
+        """Test a trivial usage of the getblocklocations RPC command."""
+        node = self.nodes[0]
+        mocktime = node.getblockheader(node.getblockhash(0))['time'] + 1
+        node.setmocktime(mocktime)
+        node.generate(7)
+
+        EXPECTED_LOCATIONS = [
+            {'file': 0, 'data': 1861, 'undo': 254, 'prev': '1893864d81638cae904357907559248964050547ddcd0dbe3f72e65b5a31bf1f'},
+            {'file': 0, 'data': 1601, 'undo': 213, 'prev': '1b086a63a5d5c320c747fee32cdb819c6f4f9eb78c24b7c5a33ae794218dc639'},
+            {'file': 0, 'data': 1341, 'undo': 172, 'prev': '448640031d1418c70a109ea4f88f497804f3336c7309ff7e535894beb80b26e4'},
+            {'file': 0, 'data': 1081, 'undo': 131, 'prev': '602403f82060e224423a9f22062fad3f5333beeee3c30313f6ea808e37b7e0b2'},
+            {'file': 0, 'data': 821, 'undo': 90, 'prev': '334ab00aba5d213c8f67161ddff346c4643d0709c3bdafa4b4b05fe6f7e4ed48'},
+            {'file': 0, 'data': 561, 'undo': 49, 'prev': '43f10598f19eced9c514f5ae40dbce0ab101362a22e18820901c6e03d7babe0b'},
+            {'file': 0, 'data': 301, 'undo': 8, 'prev': '0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206'},
+            {'file': 0, 'data': 8, 'prev': '0000000000000000000000000000000000000000000000000000000000000000'},  # genesis block
+        ]
```

Comment:
Perhaps it'd make more sense to test that the blocks stored on the returned offsets contain the same data as when using `getblock` call?

Reply:
Sounds good, will do.

Reply:
Done at https://github.com/bitcoin/bitcoin/pull/20702/commits/9d5c2bb85c2601ad54fa9df49b4f13e0490fb421#diff-dc390fc4203f5b02a7913b3f5b638b466d3fd8a14ff7fb8daa170c22f92a6dc4R51-R61.